src/librustc_resolve/lib.rs

```diff
@@ -3534,26 +3534,14 @@ fn err_path_resolution() -> PathResolution {
 }
 
 
-pub struct CrateMap {
-    pub def_map: RefCell<DefMap>,
-    pub freevars: FreevarMap,
-    pub export_map: ExportMap,
-    pub trait_map: TraitMap,
-    pub glob_map: Option<GlobMap>,
-}
-
 #[derive(PartialEq,Copy, Clone)]
 pub enum MakeGlobMap {
     Yes,
     No,
 }
 
 /// Entry point to crate resolution.
-pub fn resolve_crate<'a>(session: &'a Session,
-                         krate: &'a Crate,
-                         definitions: &'a Definitions,
-                         make_glob_map: MakeGlobMap)
-                         -> CrateMap {
+pub fn resolve_ast<'a, 'b>(resolver: &'b mut Resolver<'a>, krate: &'b Crate) {
```

Comment:
I prefer `resolve_crate`, since we are resolving the whole crate, not any part of the AST